### PR TITLE
Support pagination for ReservedInstances

### PR DESCRIPTION
**Pull request by vide**
This fixes an hideous problem: by default [DescribeReservedInstancesOfferings](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeReservedInstancesOfferings.html) returns a 100 results page, limiting how many Instance Types Edda presents to Spinnaker.

### CODE REVIEW

File: src/main/scala/com/netflix/edda/aws/AwsCrawlers.scala [hunk 1]
```diff
@@ -1057,8 +1057,20 @@ class AwsReservedInstanceCrawler(val name: String, val ctx: AwsCrawler.Context)
...
+        val response = backoffRequest { ctx.awsClient.ec2.describeReservedInstancesOfferings(request.withNextToken(this.nextToken.get)) }
+        this.nextToken = Option(response.getNextToken)
+        response.getReservedInstancesOfferings.asScala.map(
+          item => {
+            Record(item.getReservedInstancesOfferingId, ctx.beanMapper(item))
+          }).toList
+      }
+    }
+    val list = it.toList.flatten
+    list
```
Comment (brharrington) [inline src/main/scala/com/netflix/edda/aws/AwsCrawlers.scala]: This could be simplified to:

```
it.toList.flatten
```

**Comment by copperlight** [comment]
This looks reasonable and it is close to the updated version we run internally.

File: src/main/scala/com/netflix/edda/aws/AwsCrawlers.scala (hunk 1 above)
Comment (vide) [reply src/main/scala/com/netflix/edda/aws/AwsCrawlers.scala]: Yep I thought about it but @copperlight was quicker in accepting the PR. I can open another one with the improvement if you want

**Comment by vide** [comment]
Awesome! Thanks!

File: src/main/scala/com/netflix/edda/aws/AwsCrawlers.scala (hunk 1 above)
Comment (copperlight) [reply src/main/scala/com/netflix/edda/aws/AwsCrawlers.scala]: No worries, I'll just update it directly.

**Pull request merged**